resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnRMClient.scala

```diff
@@ -107,7 +107,7 @@ private[spark] class YarnRMClient extends Logging {
     // so not all stable releases have it.
     val prefix = WebAppUtils.getHttpSchemePrefix(conf)
     val proxies = WebAppUtils.getProxyHostsAndPortsForAmFilter(conf)
-    val hosts = proxies.asScala.map(_.split(":").head)
+    val hosts = proxies.asScala.map(proxy => proxy.splitAt(proxy.lastIndexOf(":"))._1)
```

Comment:
Hi, @PavithraRamachandran . If possible, could you check the code base if this is the last instance or not?

Reply:
thank you for identifying this , i shall handle such scenarios too and correct it.